flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPoolTest.java

```diff
@@ -400,6 +399,23 @@ private int getNumRequestedFromMemorySegmentPool() {
 		return networkBufferPool.getTotalNumberOfMemorySegments() - networkBufferPool.getNumberOfAvailableMemorySegments();
 	}
 
+	private BufferListener createBufferListener(int notificationTimes) {
+		return spy(new BufferListener() {
+			int times = 0;
+
+			@Override
+			public boolean notifyBufferAvailable(Buffer buffer) {
+				times++;
+				buffer.recycle();
```

Comment:
nice trick with the `recycle()` call here leading to another invocation or the release of the buffer